### ISSUE TITLE
échanges après accord commun

### ISSUE THREAD

**Issue by TerenceMarchal** [matthieu637/cpp-2a-info#30]
Depuis le début certains élèvent s'échangent des actions entres eux après s'être mis d'accord. Après étude des coups, ils font des ask/bids directement après le top. Du coup en faisant un ask moi aussi directement après le top, cela permettait de récupérer leurs actions à bas prix. Sauf que cela ne marche pas tout le temps, et M. Zimmer l'avait évoqué, tout le monde ne recoit pas le top en même temps, il arrive donc qu'ils aient le temps de faire leur échange avant que je recoive moi même le top et puisse faire un ask. Il faudrait donc trouver un moyen de synchroniser parfaitement les tops, même si ca me parait compliqué à mettre en oeuvre réellement. Si quelqu'un a une idée je suis preneur.

**Comment by matthieu637** [comment]
En effet, j'ai ensuite oublié de faire la modification.
Ça devrait aller mieux maintenant. Préviens-moi si le problème persiste encore, je laisserai un sleep le temps que les packets arrivent.

**Comment by matthieu637** [comment]
Est-ce toujours d'actualité?

**Comment by TerenceMarchal** [comment]
Désolé, je n'avais pas pu participer à la séance précédente.
Du coup je pense que le problème est toujours d'actualité, en regardant les logs voici ce que je trouve :
[['Achat', 'ValentineHognon', 'Google', 252.5, 30], ['Achat', 'MatthieuDevalle',
 'Facebook', 20.01, 1000], ['Vente', 'QuentinFabiani', 'Facebook', 20.0, 1000], 
['Achat', 'MatthieuDevalle', 'Google', 20.01, 1000], ['Vente', 'QuentinFabiani',
 'Google', 20.0, 1000], ['Achat', 'MatthieuDevalle', 'Trydea', 20.01, 1000], ['A
chat', 'QuentinFabiani', 'Trydea', 20.0, 1000], ['Achat', 'MatthieuDevalle', 'Ap
ple', 20.01, 1000], ['Achat', 'QuentinFabiani', 'Apple', 20.0, 1000], ['Achat', 
'QuentinFabiani', 'Trydea', 2.5, 2000], ['Achat', 'QuentinFabiani', 'Apple', 2.5
, 2000], ['Vente', 'MatthieuDevalle', 'Google', 249.9, 25], ['Vente', 'MatthieuD
evalle', 'Apple', 249.9, 25], ['Vente', 'MatthieuDevalle', 'Trydea', 249.9, 25],
 ['Vente', 'MatthieuDevalle', 'Facebook', 249.9, 25], ['Achat', 'DylanAlibayMand
jyDitFadjou', 'Facebook', 20.0, 1000], ['Achat', 'DylanAlibayMandjyDitFadjou', '
Google', 20.0, 1000], ['Vente', 'DylanAlibayMandjyDitFadjou', 'Apple', 20.0, 100
0], ['Vente', 'CharlotteStutzmann', 'Facebook', 166.66667, 1], ['Vente', 'DylanA
libayMandjyDitFadjou', 'Trydea', 20.0, 1000], ['Vente', 'CharlotteStutzmann', 'A
pple', 166.66667, 1], ['Achat', 'DylanAlibayMandjyDitFadjou', 'Facebook', 5.0, 1
000], ['Vente', 'CharlotteStutzmann', 'Google', 166.66667, 1], ['Achat', 'DylanA
libayMandjyDitFadjou', 'Google', 5.0, 1000], ['Vente', 'CharlotteStutzmann', 'Tr
ydea', 166.66667, 1], ['Achat', 'TerenceMarchal', 'Apple', 25.0, 1000], ['Achat'
, 'TerenceMarchal', 'Facebook', 25.0, 1000], ['Achat', 'TerenceMarchal', 'Google
', 25.0, 1000], ['Achat', 'TerenceMarchal', 'Trydea', 25.0, 1000],
Plusieurs joueurs ont le temps d'effectuer plusieurs ordres d'achats ou de vente avant moi, alors que directement après le top je ne fais qu'un r.solde() avant d'effectuer mes 4 ordres d'achats. Je ne pense pas que ce délai soit suffisant pour que les autres joueurs puissent effectuer autant de requêtes (Matthieu effectue 8 ordres avant moi).
Cordialement.

**Comment by matthieu637** [comment]
Est-ce que tu joues sans interface graphique? Es-tu en connexion wifi ? Est-ce qu'ils sont sur la VM?
Je vais augmenter le délai de 2 secondes à 3.5 secondes pour le lancement.

**Comment by TerenceMarchal** [comment]
Effectivement, je joue avec interface graphique (mais sans l'affichage des courbes), en wifi et sans la VM, ce qui n'arrange pas les choses. @MatthieuDEVALLE t'étais sur la VM ? Et t'as toujours l'interface graphique non ?
J'essayerais la prochaine fois sans interface et depuis la VM, j'avais totalement oublié ces aspects, ca devrait aller mieux.
Enfin, le jour du ds, il faudrait vérifier que tout le monde ai bien le client.py original, puisqu'il est facile de passer outre les sécurités que vous avez mis en place en modifiant quelques lignes (plusieurs élèves l'ont remarqué).

**Comment by MatthieuDEVALLE** [comment]
Oui, je suis sur la VM avec une interface graphique (un peu réduite) et je passe tous mes ordres directement après le r.top().

**Comment by matthieu637** [comment]
De quelle sécurité tu parles ? 
S'il s'agit du sleep, les étudiants peuvent toujours modifier le client.py (en retirant le sleep) ce n'est pas pour autant que le serveur acceptera leurs requêtes avant.
